lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -595,4 +598,102 @@
         }
         return o;
     });
+
+    platform.registerFunction(platform.FunctionKind.Math_min, function (value1, value2) {
+        // #sec-math.min
+
+        // If no arguments are given, the result is positive infinity
+        // If any value is NaN, the result is NaN. 
+        // The comparison of values to determine the smallest value is done using the Abstract Relational Comparison algorithm except that +0 is considered to be larger than -0.
+        if (arguments.length === 0 ) {
+            return __chakraLibrary.positiveInfinity;
+        }
+        
+        value1 = +value1;
+        if (value1 !== value1) {
+            return NaN;
+        }
+
+        if (arguments.length === 1) {
+            return value1;
+        }
+
+        value2 = +value2;
+        if (value2 !== value2) {
+            return NaN;
+        }
+
+        if (arguments.length === 2) {
+            if ((value1 < value2) || (value1 === value2 && value1 === 0 && 1/value1 < 1/value2)) { // checks for -0 and +0
+                return value1;
+            }
+            else {
+                return value2;
+            }
+        }
+
+        let min = value1;
+        let nextVal;
+
+        for (let i = 1; i < arguments.length; i++) {
+            nextVal = +arguments[i];
+            if (nextVal !== nextVal) {
+                return NaN;
+            }
+            else if ((min > nextVal) || (min === nextVal && min === 0 && 1/min > 1/nextVal)) { // checks for -0 and +0
+                min = nextVal;
+            }
+        }
+        
+        return min;
+    });
+
+    platform.registerFunction(platform.FunctionKind.Math_max, function (value1, value2) {
+        // #sec-math.max
+
+        // If no arguments are given, the result is negative infinity
+        // If any value is NaN, the result is NaN. 
+        // The comparison of values to determine the largest value is done using the Abstract Relational Comparison algorithm except that +0 is considered to be larger than -0.
+        if (arguments.length === 0) {
+            return __chakraLibrary.negativeInfinity;
+        }
+        
+        value1 = +value1;
+        if (value1 !== value1) {
+            return NaN;
+        }
+
+        if (arguments.length === 1) {
+            return value1;
+        }
+
+        value2 = +value2;
+        if (value2 !== value2) {
+            return NaN;
+        }
+
+        if (arguments.length === 2) {
+            if ((value1 > value2) || (value1 === value2 && value1 === 0 && 1/value1 > 1/value2)) { // checks for -0 and +0
+                return value1;
+            }
+            else {
+                return value2;
+            }
+        }
+
+        let max = value1;
+        let nextVal;
+
+        for (let i = 1; i < arguments.length; i++) {
+            nextVal = +arguments[i];
```

Comment:
Ready to get super pedantic? Consider the following: `i=0;o={valueOf:()=>{return ++i}};print(Math.max(o,o,o))`. Current implementation returns 3; this implementation would return 4 because it calls ToNumber on the second argument twice. I think you should move the block `value2 = +value2; if (value2 !== value2) { return NaN; }` inside the `if (arguments.length === 2)` part. Same goes for min of course.

Reply:
1a82165a5e48c87f6559f5c109d3810c62b0800e